python/cudf/cudf/core/reshape.py

```diff
@@ -1167,6 +1167,8 @@ def pivot(
     # MultiIndex to Index
     if not values_is_list:
         result._data.droplevel(0)
+        if cudf.get_option("mode.pandas_compatible"):
+            result._data.label_dtype = column_data.dtype
```

Comment:
Shouldn't `_pivot` be preserving the `label_dtype` correctly regardless of pandas compat mode?

Reply:
Done.